mods/battle-royale/hook/lua/defaultunits.lua

```diff
@@ -0,0 +1,229 @@
+local oldACU = ACUUnit
+local import = import
+
+local TransferUnitsOwnership = import('/lua/SimUtils.lua').TransferUnitsOwnership
+local EffectTemplates = import("/lua/EffectTemplates.lua")
+local EffectUtils = import("/lua/EffectUtilities.lua")
+local CarePackageTeleportIn = EffectTemplates.UnitTeleport01
+local CarePackageDestroyed = table.concatenate(EffectTemplates.CommanderQuantumGateInEnergy,EffectTemplates.AGravitonBolterHit01)
+local CarePackageOnWater = EffectTemplates.DefaultSeaUnitBackWake01
+
+
+
+--if multiplier = 0.75
+local T2StageTime = 720 -- 12 minute
+local T3StageTime = 1500 -- 25 minute
+local Multiplier = ScenarioInfo.Options.CarePackagesCurve
+local TargetArmy = ScenarioInfo.Options.TargetArmy
+
+--- Initializes the time at which the commander's beacon will enter the t2 and t3 stages.
+--- Depends on the CarePackagesCurve parameter defined in the lobby.
+function InitStageTime()
+    if Multiplier == 1.0 then
+        T2StageTime = 630 -- 10 minute 30 second
+        T3StageTime = 1320 -- 22 minute
+    end
+
+    if Multiplier == 1.25 then
+        T2StageTime = 540 --9 minute
+        T3StageTime = 1000 -- 18 minute 20 second
+    end
+
+    if Multiplier == 1.5 then
+        T2StageTime = 420 -- 7 minute
+        T3StageTime = 840 -- 14 minute
+    end
+end
+
+InitStageTime()
+
+--- Depending on the time elapsed in the game, returns the id t1, t2 or t3 of the commander's beacon.
+function DefineBeaconId()
+    local result = 'uac1301'
+    local inGameTime = GetGameTimeSeconds()
+
+    if inGameTime > T2StageTime and inGameTime < T3StageTime then
+        result = 'xsc1501'
+    end
+
+    if inGameTime > T3StageTime then
+        result = 'xsc1301'
+    end
+
+    return result
+end
+
+function ResetUnitOrders(units)
```

Comment:
There is no description because the name of the function fully reflects the functionality and the function does not contain a non-obvious implementation.